packages/api/src/auth/decoders/auth0.ts

```diff
@@ -1,5 +1,45 @@
 import jwt from 'jsonwebtoken'
-import jwksClient from 'jwks-rsa'
+import jwksClient, { JwksError, SigningKeyNotFoundError } from 'jwks-rsa'
+
+export const auth0Config = () => {
+  const { AUTH0_DOMAIN, AUTH0_AUDIENCE } = process.env
+
+  if (!AUTH0_DOMAIN || !AUTH0_AUDIENCE) {
+    throw new Error('`AUTH0_DOMAIN` or `AUTH0_AUDIENCE` env vars are not set.')
+  }
+
+  return {
+    issuer: `https://${AUTH0_DOMAIN}`,
+    audience: AUTH0_AUDIENCE,
+  }
+}
+
+const auth0Client = () => {
+  return jwksClient({
+    jwksUri: `${auth0Config().issuer}/.well-known/jwks.json`,
+  })
+}
+
+export const getAuth0SigningKey = async (header: any) => {
+  const kid = header.kid as string
+
+  try {
+    return await auth0Client().getSigningKey(kid)
+  } catch (error) {
+    console.error(error)
+    throw new SigningKeyNotFoundError(kid)
+  }
+}
```

Comment:
It looks like `getSigningKey` already throws this error if it can't find a signing key that matches `kid`: https://cs.github.com/auth0/node-jwks-rsa/blob/e7317dd7eeae6244c68fa3220a88dcf271b44ec9/src/JwksClient.js#L84